### PR TITLE
grandorgue: 0.3.1-r2333 -> 3.8.0-1, add darwin support

### PR DESCRIPTION
**Pull request by wegank**
###### Description of changes

https://github.com/GrandOrgue/grandorgue/releases/tag/3.8.0-1

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/audio/grandorgue/default.nix [hunk 1]
```diff
@@ -1,32 +1,56 @@
...
-    inherit rev;
-    sha256 = "0xzjdc2g4gja2lpmn21xhdskv43qpbpzkbb05jfqv6ma2zwffzz1";
+  version = "3.8.0-1";
+
+  src = fetchFromGitHub {
+    owner = "GrandOrgue";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-VXf2B5NK6lrcNXUiTTjYhfBVrTWusyadD+5ySkmelsI=";
+    fetchSubmodules = true;
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/grandorgue/default.nix:15]:
```suggestion
    fetchSubmodules = true;
    sha256 = "sha256-VXf2B5NK6lrcNXUiTTjYhfBVrTWusyadD+5ySkmelsI=";
```
fetchSubmodules influeneces the hash

File: pkgs/applications/audio/grandorgue/default.nix [hunk 2]
```diff
@@ -1,32 +1,56 @@
...
   };
 
-  nativeBuildInputs = [ cmake pkg-config ];
+  postPatch = ''
+    substituteInPlace resources/CMakeLists.txt --replace \
+      "iconutil -c icns \''${GENERATED_ICONS_DIR}" \
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/grandorgue/default.nix:20]:
```suggestion
    substituteInPlace resources/CMakeLists.txt \
      --replace "iconutil -c icns \''${GENERATED_ICONS_DIR}" \
```

File: pkgs/applications/audio/grandorgue/default.nix [hunk 3]
```diff
@@ -1,32 +1,56 @@
...
   ] ++ lib.optional (!includeDemo) "-DINSTALL_DEMO=OFF";
 
+  NIX_CFLAGS_COMPILE = lib.optional stdenv.isDarwin "-DTARGET_OS_IPHONE=0";
+
+  postInstall = lib.optionalString stdenv.isDarwin ''
+    mkdir -p $out/{Applications,bin,lib}
+    mv $out/GrandOrgue.app $out/Applications/
+    for lib in $out/Applications/GrandOrgue.app/Contents/MacOS/lib*; do
+      ln -s $lib $out/lib/
+    done
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/grandorgue/default.nix:45]: We can't just symlink the entire lib directory?

File: pkgs/applications/audio/grandorgue/default.nix [hunk 4]
```diff
@@ -1,32 +1,56 @@
...
+    makeWrapper $out/{Applications/GrandOrgue.app/Contents/MacOS,bin}/GrandOrgue
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/grandorgue/default.nix:46]:
```suggestion
    ln -s $out/Applications/GrandOrgue.app/Contents/MacOS/GrandOrgue $out/bin/GrandOrgue
```

File: pkgs/applications/audio/grandorgue/default.nix (hunk 1 above)
Comment (wegank) [reply pkgs/applications/audio/grandorgue/default.nix]: Done.

File: pkgs/applications/audio/grandorgue/default.nix (hunk 3 above)
Comment (wegank) [reply pkgs/applications/audio/grandorgue/default.nix:46]: Nope:

```bash
$ ls result/Applications/GrandOrgue.app/Contents/MacOS
GrandOrgue                      GrandOrgueTool                  libGrandOrgueImages.3.7.2.dylib
GrandOrguePerfTest              libGrandOrgueCore.3.7.2.dylib
```

File: pkgs/applications/audio/grandorgue/default.nix (hunk 4 above)
Comment (wegank) [reply pkgs/applications/audio/grandorgue/default.nix:47]: This prevents `GrandOrgue` from finding resources, so a `makeWrapper` is needed.

**Pull request merged**